### PR TITLE
Secrets: Add fallback to secrets kvstore plugin

### LINKED ISSUE #52743: Chore: Refactor secret plugin migration to allow for cleaner unit testing
In order to test an edge case around migration of secrets to the AWS plugin, I had to create a slightly hacky function to allow us to override some behavior within a function for our unit test. This can be found in the test `TestFatalPluginErr_MigrationTestWithErrorDeletingUnifiedSecrets` in the call to `PluginSecretMigrationService.overrideGetAllFunc`. 

This can be made cleaner by allowing the Migrate function to take override options instead, and using this to provide a mock SQL Secrets store that returns the values we need.

### PR DESCRIPTION
**Pull request by guicaulada**
**What this PR does / why we need it**:

This PR adds a fallback SecretKVStore to the plugin that can be added or removed, in this PR the fallback is not being used anywhere except on migration, this fallback allows us to remove the override function needed only for testing purposes while at the same time making it possible for us to improve Grafana's startup speed in the future by converting secret store plugin migrations to a background service.

**Which issue(s) this PR fixes**:

Fixes #52743 and #52564

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/secrets/kvstore/kvstore.go [hunk 1]
```diff
@@ -80,6 +81,9 @@ type SecretsKVStore interface {
 	Del(ctx context.Context, orgId int64, namespace string, typ string) error
 	Keys(ctx context.Context, orgId int64, namespace string, typ string) ([]Key, error)
 	Rename(ctx context.Context, orgId int64, namespace string, typ string, newNamespace string) error
+	GetAll(ctx context.Context) ([]Item, error)
+	Fallback() SecretsKVStore
```
Comment (mmandrus) [inline pkg/services/secrets/kvstore/kvstore.go:85]: What is your opinion on using a combined getter/setter here in order to avoid making the interface too bulky?

Interface would look like:
```
Fallback(fb SecretsKVStore) SecretsKVStore
```

Impl would look like:
```
func (kv *SecretsKVStore) Fallback(fb SecretsKVStore) SecretsKVStore {
    if fb != nil {
        kv.fallback = fb
    }
    return kv.fallback
}
```

**Review by mmandrus — CHANGES_REQUESTED** [review]

File: pkg/services/secrets/kvstore/kvstore.go (hunk 1 above)
Comment (guicaulada) [reply pkg/services/secrets/kvstore/kvstore.go:85]: I was planning on keeping them separate so we can SetFallback(nil) to disable it.

I think it's a bit weird to have a getter-setter function on Golang since there are no default arguments, in TypeScript it would make sense since Fallback() would be the getter, Fallback(null) would be the "unsetter" and Fallback(fb) could be a setter, but in Golang we would have to use Fallback(nil) as a getter and Fallback(fb) as a setter, needing another function to work as "unsetter" like Remove... or DisableFallback() so in the end we would still have 2 functions.

Comment (mmandrus) [reply pkg/services/secrets/kvstore/kvstore.go:85]: Yeah, that makes sense.

**Review by mmandrus — APPROVED** [review]
lgtm

File: pkg/services/secrets/kvstore/kvstore.go [hunk 2]
```diff
@@ -62,6 +62,7 @@ func ProvideService(
 				log:                            logger,
 				kvstore:                        namespacedKVStore,
 				backwardsCompatibilityDisabled: features.IsEnabled(featuremgmt.FlagDisableSecretsCompatibility),
+				fallback:                       store,
```
Comment (leandro-deveikis) [inline pkg/services/secrets/kvstore/kvstore.go:65]:
```suggestion
				// as the plugin is installed, secretsKVStoreSQL is now replaced with an instance of secretsKVStorePlugin with the sql store as a fallback (used for migration and in case a secret is not found). 
				fallback:                       store,
```

This part took me a while to follow correctly, I am just suggesting adding a similar comment for the future

**Review by leandro-deveikis — APPROVED** [review]
LGTM! Nice work! :tada: 
I just left a comment about adding a comment :P but is just optional
